### PR TITLE
⚡ THU-321: Fix stale "last synced at" timestamp

### PR DESCRIPTION
**Pull request by cjroth**
## Summary
- Track download completion transitions as a local `lastSyncedAt` timestamp, since PowerSync's native `lastSyncedAt` only updates on full checkpoint syncs and can appear hours stale during incremental sync activity
- Add a 30-second periodic timer so the elapsed time display ("Synced Xm ago") stays fresh between sync status events
- Extract `formatSyncElapsed` and `mostRecentDate` as testable pure functions with unit tests

## Linear
[THU-321](https://linear.app/mozilla-thunderbolt/issue/THU-321/last-synced-at-timestamp-is-sometimes-not-updated)

## Test Plan
- [ ] Verify "last synced at" updates after each sync cycle (not just full checkpoints)
- [ ] Verify elapsed time text transitions from "Just synced" → "Synced 1m ago" → etc. over time
- [ ] Verify no regressions in sync status indicator (connecting, offline, disabled states)
- [ ] Unit tests pass: `bun test src/hooks/use-powersync-status.test.ts src/components/powersync-status.test.ts`

## Changes
- `src/hooks/use-powersync-status.ts` — Track download completion via ref-based state to provide a responsive `lastSyncedAt`; export `mostRecentDate` utility
- `src/components/powersync-status.tsx` — Add periodic timer for elapsed time refresh; extract `formatSyncElapsed` function
- `src/hooks/use-powersync-status.test.ts` — Tests for `mostRecentDate`
- `src/components/powersync-status.test.ts` — Tests for `formatSyncElapsed`

---

> [!NOTE]
> **Medium Risk**
> Adjusts sync-status logic and introduces a periodic UI timer, which could change when/what users see for “last synced” and has potential for edge-case regressions in status transitions.
> 
> **Overview**
> **Fixes stale “last synced” UI by making `lastSyncedAt` update on incremental sync activity.** The PowerSync status hook now derives a *local* `lastSyncedAt` by detecting `downloading` transitions (true → false) and choosing the most recent timestamp vs PowerSync’s reported `lastSyncedAt`.
> 
> **Keeps elapsed-time text fresh without new status events.** The header indicator extracts `formatSyncElapsed(...)`, adds a 30s interval tick (and an immediate refresh when `lastSyncedAt` changes), and includes new unit tests covering the elapsed-time formatting edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ce85f0492e2042c2b0486935f4be8d34cb69f522. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/hooks/use-powersync-status.ts [hunk 1]
```diff
@@ -19,10 +19,30 @@ export type PowerSyncStatusInfo = {
   lastSyncedAt: Date | null
 }
 
+/**
```
Comment (cjroth) [inline src/hooks/use-powersync-status.ts]: Is there not a dayjs way to do this? And if not, should this go into a util file?

**Comment by cjroth** [comment]
⚡ Addressed in the latest push — moved mostRecentDate to shared utils and switched to dayjs.isAfter() for consistency.

**Review by raivieiraadriano92 — APPROVED** [review]

**Review by raivieiraadriano92 — COMMENTED** [review]
I'd say these changes are no longer required - I double check that and it seems to be working properly in prod .

That`lastSyncedAt` that we use comes from the PowerSyncDatabase instance - PowerSync automatically updates that one.

As you mentioned on Notion, this fixed something, but I'm not sure what it had fixed exactly.

**Comment by cjroth** [comment]
Closing based on @raivieiraadriano92 's comment.
